src/contracts/proxies/NFTRenderer.sol

```diff
@@ -114,13 +156,9 @@ contract NFTRenderer {
     bytes32 cType;
     // scoped to reduce call stack
     {
-      IODSafeManager.SAFEData memory safeMangerData = _safeManager.safeData(_safeId);
-      address safeHandler = safeMangerData.safeHandler;
-      cType = safeMangerData.collateralType;
-
-      ISAFEEngine.SAFE memory SafeEngineData = _safeEngine.safes(cType, safeHandler);
-      uint256 collateral = SafeEngineData.lockedCollateral;
-      uint256 debt = SafeEngineData.generatedDebt;
+      uint256 collateral;
+      uint256 debt;
+      (cType, collateral, debt) = getVaultCTypeAndCollateralAndDebt(_safeId);
```

Comment:
Excellent decision to simplify and remove duplicate code. However, the NFTRenderer was intentionally isolated from the rest of the 721 contract, in order to reduce the security implications of making upgrades. I believe we should continue to keep it limited to only SVG image functionality.

I am marking approved, and we can continue the discussion in #328 for isolating parts of the state-hash feature to its own contract.